### PR TITLE
[INFRA/IIGO] Removing Globals from IIGO : Part 4/4 ... maybe more

### PR DESCRIPTION
**Pull request by preetl**
# Summary

Moved IIGOSanctionCache and IIGOHistoryCache to game state. 
Building up to removing monitoring. Sorry about LOC, had to move some structs because of an import cycle.

## Additional Information

N/A

## Test Plan

Still passing!

### CODE REVIEW

File: internal/clients/team3/judge.go [hunk 1]
```diff
@@ -22,15 +21,15 @@ func (j *judge) PayPresident() (shared.Resources, bool) {
...
+func (j *judge) InspectHistory(iigoHistory []shared.Accountability, turnsAgo int) (map[shared.ClientID]shared.EvaluationReturn, bool) {
+	outMap := map[shared.ClientID]shared.EvaluationReturn{}
 
 	// If we do not have sufficient budget to conduct the inspection,
 	// then we will return an empty map with true evaluations.
 	// if j.c.getLocalResources() < config.IIGOConfig.InspectHistoryActionCost {
 	// 	// dummy evaluation map
 	// 	for _, entry := range iigoHistory {
-	// 		outMap[entry.ClientID] = roles.EvaluationReturn{
+	// 		outMap[entry.ClientID] = shared.EvaluationReturn{
```
Comment (Neelesh99) [inline internal/clients/team3/judge.go:32]: Wait why do we have this random commented block XD

File: internal/common/shared/sanctions.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
+package shared
```
Comment (Neelesh99) [inline internal/common/shared/sanctions.go:1]: This stuff makes more sense here tbf

File: internal/server/iigointernal/sanctionevaluator.go [hunk 3]
```diff
@@ -7,6 +7,24 @@ import (
 	"gonum.org/v1/gonum/mat"
 )
 
+// DefaultInitLocalSanctionCache generates a blank sanction cache
+func DefaultInitLocalSanctionCache(depth int) map[int][]shared.Sanction {
```
Comment (Neelesh99) [inline internal/server/iigointernal/sanctionevaluator.go:11]: Why move and export these if we aren't using them in any other package?

File: internal/server/server.go [hunk 4]
```diff
@@ -74,12 +74,14 @@ func createSOMASServer(
...
-			Environment:     disasters.InitEnvironment(clientIDs, gameConfig.DisasterConfig),
-			ForagingHistory: forageHistory,
-			IIGOHistory:     map[uint][]shared.Accountability{},
+			Season:            1,
+			Turn:              1,
+			ClientInfos:       clientInfos,
+			Environment:       disasters.InitEnvironment(clientIDs, gameConfig.DisasterConfig),
+			ForagingHistory:   forageHistory,
+			IIGOHistory:       map[uint][]shared.Accountability{},
+			IIGOSanctionCache: map[int][]shared.Sanction{},
```
Comment (Neelesh99) [inline internal/server/server.go]: Where are these initialised?

**Review by Neelesh99 — CHANGES_REQUESTED** [review]
I think if you use the exported function to initialise the caches in server.go we should be ok.

File: internal/server/server.go (hunk 4 above)
Comment (preetl) [reply internal/server/server.go]: aaaaa

File: internal/server/iigointernal/sanctionevaluator.go (hunk 3 above)
Comment (preetl) [reply internal/server/iigointernal/sanctionevaluator.go:11]: AAAAAA

File: internal/clients/team3/judge.go (hunk 1 above)
Comment (preetl) [reply internal/clients/team3/judge.go:32]: lol i have no idea, shall i uncomment?

File: internal/server/iigointernal/judiciary.go [hunk 5]
```diff
@@ -47,11 +45,11 @@ func (j *judiciary) syncWithGame(gameState *gamestate.GameState, gameConf *confi
 }
 
 func (j *judiciary) resetCaches() {
-	if len(j.localSanctionCache) != int(j.gameConf.SanctionCacheDepth) {
```
Comment (Neelesh99) [inline internal/server/iigointernal/judiciary.go:50]: Wait, this shouldn't be used every round

**Comment by preetl** [comment]
> I think if you use the exported function to initialise the caches in server.go we should be ok.

Good spot! Done!

File: internal/server/iigointernal/judiciary.go (hunk 5 above)
Comment (Neelesh99) [reply internal/server/iigointernal/judiciary.go:50]: I'll raise a PR for this, and a riot

File: internal/server/iigointernal/sanctionevaluator.go (hunk 3 above)
Comment (Neelesh99) [reply internal/server/iigointernal/sanctionevaluator.go:11]: Nice

**Review by Neelesh99 — APPROVED** [review]
Changes made LGTM.

**Comment by lhl2617** [comment]
we still got monitoring btw :D this lgtm however

**Review by lhl2617 — APPROVED** [review]
gogogogo

File: internal/common/shared/sanctions.go [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+// Provides enumerated tiers for IIGO sanctions
+const (
+	SanctionTier1 IIGOSanctionsTier = iota
+	SanctionTier2
+	SanctionTier3
+	SanctionTier4
+	SanctionTier5
+	NoSanction
+)
```
Comment (lhl2617) [inline internal/common/shared/sanctions.go:24]: ah missed this--can you do the `String` and `Marshal` functions as per #150?

**Review by lhl2617 — CHANGES_REQUESTED** [review]
as per comment

File: internal/common/shared/sanctions.go (hunk 6 above)
Comment (lhl2617) [reply internal/common/shared/sanctions.go:24]: (if this is inside gamestate)

Comment (preetl) [reply internal/common/shared/sanctions.go:24]: sure thing! done :)

**Review by lhl2617 — APPROVED** [review]
GOGOGGOOGOGO

**Pull request merged**